### ISSUE TITLE
Add `StandardScaling` Transformer

### ISSUE THREAD

**Issue by sarahmish** [sdv-dev/RDT#172]
### Problem Description

Create a new Transformer for numerical values that scales them via standard normalization (z-score). 

```python3
class StandardTransformer:
    def __init__(self):
        pass

    def fit(self, data):
        # learn mean and standard deviation values
        pass

    def transform(self, data):
        # data - mean(data) / std(data) - min(data))
        pass

    def reverse_transform(self, data):
        pass

```

### Expected behavior

Example case
```python3
>>> data = [-1, 1, 3]
>>> transformer = StandardTransformer()
>>> transformer.fit(data)
>>> transformer.transform(data)
[-1.224 0. 1.224]
>>> transformer.reverse_transform(transformer.transform(data))
[-1. 1. 3.]
```

### Additional context

Possibly similar to scikit-learn's [StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html#sklearn.preprocessing.StandardScaler)

**Comment by npatki** [comment]
It seems that using the [GaussianNormalizer](https://docs.sdv.dev/rdt/transformers-glossary/numerical/gaussiannormalizer) with distribution set to `'norm'` will accomplish exactly this: It will estimate the original distribution as a normal distribution by learning mean/std). Then, it will transform the data into a standard normal distribution with mean=0, std=1 (z-score).
